### PR TITLE
Login validacion 1

### PR DESCRIPTION
**Pull request by lfmontesanto**
Summary
*App mounted over Bootstrap
*Invalid credentials warnings
*Credentials stored in order to avoid logging in again once the app is closed

Screenshot
-
screenshot from 2018-10-12 17-28-15
screenshot from 2018-10-12 17-28-25
screenshot from 2018-10-12 17-28-37

Trello Card
https://trello.com/c/6hbmhFfp

### CODE REVIEW

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/RootActivity.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class RootActivity extends WolmoActivity {
+    @Override
+    protected int layout() {
+        return  R.layout.activity_base;
+    }
+
+    @Override
+    protected void init() {
+        Intent intent;
+        SharedPreferences pref = getSharedPreferences("prefs", Context.MODE_PRIVATE);
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/RootActivity.java]: Try to move the hardcode strings to the xml file or create a constant

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/RootActivity.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        String username = pref.getString("mail", "");
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/RootActivity.java]: Same here

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/RootActivity.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        if (username.equals("")){
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/RootActivity.java]: You can use the [isEmpty](https://www.geeksforgeeks.org/java-string-isempty-method-example/) of the string object

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    public int layout() {
+        return R.layout.fragment_home;
+    }
+
+    @Override
+    public void init() {
+
+    }
+    /*
+    @OnClick(R.id.logOut)
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.java]: It's ok in this case but don't leave commented code when you create a PR

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/HomePresenter.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+
+class HomePresenter extends BasePresenter<IHomeView> {
+    @Inject
+    public HomePresenter(){
+
+    }
+    public void doLogOut( Context context) {
+        SharedPreferences sharedPref = context.getSharedPreferences("prefs",Context.MODE_PRIVATE);
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/HomePresenter.java]: Same here with the hardcode strings

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    public void init() {
+    }
+
+    private boolean validarEmail(String email){
+        return  !TextUtils.isEmpty(email) && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
+    }
+    private boolean validarPassword (String password){
+        Pattern pattern;
+        Matcher matcher;
+        final String PASSWORD_PATTERN = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9]).{8,}$";
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java:56]: checking the password it's not required. But next time you can create a new Class like PatternMatcherHelper or something like that , that does the work.

File: Bootstrap/app/src/main/res/layout/fragment_home.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">>
+
+    <android.support.v7.widget.Toolbar
+        android:id="@+id/toolbar2"
```
Comment (DemianIac) [inline Bootstrap/app/src/main/res/layout/fragment_home.xml]: try to use other names like fragment_home_toolbar

File: Bootstrap/app/src/main/res/layout/fragment_home.xml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="56dp"
```
Comment (DemianIac) [inline Bootstrap/app/src/main/res/layout/fragment_home.xml]: don't leave hardcode dimens , use the xml resource for that

File: Bootstrap/app/src/main/res/layout/fragment_home.xml [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        android:textAlignment="viewStart"
+        android:textAllCaps="false"
+        android:textColor="@android:color/black"
+        android:textSize="24sp"
+        android:textStyle="bold"
+        android:visibility="visible" />
+
+    <android.support.v7.widget.Toolbar
+        android:id="@+id/toolbar3"
+        android:layout_width="403dp"
```
Comment (DemianIac) [inline Bootstrap/app/src/main/res/layout/fragment_home.xml]: Same here

File: Bootstrap/app/src/main/res/layout/fragment_home.xml [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        android:background="@android:color/background_light"
+        android:minHeight="?attr/actionBarSize"
+        android:theme="?attr/actionBarTheme"
+        android:translationY="56dp" />
+
+    <ImageView
+        android:id="@+id/imageView4"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="72dp"
```
Comment (DemianIac) [inline Bootstrap/app/src/main/res/layout/fragment_home.xml]: Same Here

File: Bootstrap/app/src/main/res/layout/fragment_login.xml [hunk 11]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DemianIac) [inline Bootstrap/app/src/main/res/layout/fragment_login.xml:2]: Check the names and hardcode dimes of this file

**Review by DemianIac — CHANGES_REQUESTED** [review]
Great work Facu!. Just some observations 

-Don't leave commented code on the files.
-Use better names for the views like fragment_home_*. This is going to help other developers to understand and follow your code.
-Avoid hardcode strings and dimens . Use the resources , that will help you to changes values in the future with no effort.

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+            if (validarEmail(emailText.getText().toString())) {
+                if (validarPassword(passwordText.getText().toString())) {
+                    getPresenter().doLogIn(emailText.getText().toString(),passwordText.getText().toString(),getContext());
+                    Intent intent=new Intent(getActivity(),HomeActivity.class);
+                    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                    getActivity().startActivity(intent);
+                    getActivity().finish();
+                }
+                else
+                    passwordText.setError("Invalid Password");
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java]: The text should be in the strings xml. You can get the resource here and use it

File: Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+            }
+            else
+                emailText.setError("Invalid Email");
```
Comment (DemianIac) [inline Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java]: same here

**Review by DemianIac — CHANGES_REQUESTED** [review]
Just 2 hardcode strings !!. So close :P

**Comment by lfmontesanto** [comment]
Summary: 
*All hard-coded fields were replaced 
*SignUp blank activity created and linked to the sign up button
*terms&conditions url added and working on click
* **gradient added and visible during preview but not when running the app**
Trello card: 
https://trello.com/c/EFnW5Xmr

**Comment by DemianIac** [comment]
@lfmontesanto You can take pictures in the logcat tab of the AS

screen shot 2018-10-16 at 17 10 54

**Comment by lfmontesanto** [comment]
Summary:
*All hard-coded fields were replaced
**gradient added and visible during preview but not when running the app**
screenshot_1539722487
screenshot_1539722490
screenshot_1539722504
screenshot_1539722529




Trello card:
https://trello.com/c/EFnW5Xmr

**Review by DemianIac — APPROVED** [review]
👍

**Pull request merged**